### PR TITLE
refactor: replace anyhow with eyre

### PR DESCRIPTION
**Pull request by prestwich**
## Motivation

As foundry is using eyre, we should transition more crates to use it instead of anyhow

## Solution

Move the contract abigen crate to eyre

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: ethers-contract/ethers-contract-abigen/src/contract.rs [hunk 1]
```diff
@@ -199,7 +199,7 @@ impl Context {
             };
 
             if method_aliases.insert(signature.clone(), alias).is_some() {
-                return Err(anyhow!("duplicate method signature '{}' in method aliases", signature,))
+                return Err(eyre!("duplicate method signature '{}' in method aliases", signature,))
```
Comment (mattsse) [inline ethers-contract/ethers-contract-abigen/src/contract.rs:202]:
```suggestion
              eyre::bail!("duplicate method signature '{}' in method aliases", signature)
```

**Review by mattsse — APPROVED** [review]
lgtm!

File: ethers-contract/ethers-contract-abigen/src/contract.rs (hunk 1 above)
Comment (prestwich) [reply ethers-contract/ethers-contract-abigen/src/contract.rs]: fixed several places, thanks

**Comment by prestwich** [comment]
cool, I just reviewed all eyre uses in the lib and made them more idiomatic with `bail!` and `ensure!` whereever possible

**Review by gakonst — APPROVED** [review]
gg

**Pull request merged**